### ISSUE TITLE
Outpost Swarm, error while loading

### ISSUE THREAD

**Issue by thenickman100** [ruffle-rs/ruffle#18451]
### Describe the bug

An error message pops up in the console during loading and the screen remains black.  This game works fine in Flash Player, but does not load in Ruffle.  I have tried both the self-hosted version and the Ruffle SWF Player extension with no success.

### Expected behavior

The game should load.

### Content Location

https://archive.org/details/outpost-flash-game-series

[604534_OutpostSwarm.swf](https://archive.org/download/outpost-flash-game-series/604534_OutpostSwarm.swf)

[OutpostSwarmMTY3Nw==.swf](https://archive.org/download/outpost-flash-game-series/OutpostSwarmMTY3Nw%3D%3D.swf)

[outpostswarm_v1.0.swf](https://archive.org/download/outpost-flash-game-series/outpostswarm_v1.0.swf)

### Affected platform

Self-hosted version

### Operating system

Windows 11

### Browser

Microsoft Edge Version 130.0.2849.56 (Official build) (64-bit)

### Additional information

ERROR core/src/avm2/events.rs:432 Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x166d280 }) to handler FunctionObject(FunctionObject { ptr: 0x131cd18, name: Ok("Preloader/mainloop3()") }) : ArgumentError: Error #2025: The supplied DisplayObject must be a child of the caller.

**Comment by Lord-McSweeney** [comment]
The game works for me on the latest desktop Ruffle, are you sure you're testing with the latest version of Ruffle?

**Comment by thenickman100** [comment]
Yes, I am using today's version of the self-hosted ruffle.  Outpost Haven is working fine, but Outpost Swarm is not.  Here are the different errors I get from each of the different swf files.  I'm trying to run while offline as well.

![Image](https://github.com/user-attachments/assets/87de8e1e-5521-453b-bab5-4c9f3e782caf)
![Image](https://github.com/user-attachments/assets/1f937c0d-21b1-4c2e-8424-fb878766df11)
![Image](https://github.com/user-attachments/assets/7ee2240f-c638-40e5-808d-f28223292d9d)

**Comment by thenickman100** [comment]
Still getting the same error when trying to load this game

<img width="1332" height="281" alt="Image" src="https://github.com/user-attachments/assets/bcbff38a-e51d-4c10-bda5-5928597e9ee5" />

Any idea what is causing this issue and whether it is an easy fix?  I tried running it in the ruffle demo as well with no success.

**Comment by Lord-McSweeney** [comment]
Can you provide a link to the site you're playing the game on?

**Comment by thenickman100** [comment]
> Can you provide a link to the site you're playing the game on?


https://ruffle.rs/demo/



Alternatively, this is where I took the specific error screenshot, though I turned off the trace messages since then: https://hackahobo.info/games.php?game=outpostswarm

**Comment by Lord-McSweeney** [comment]
Have you tried playing the game on that site using the official Flash Player? From what I can tell, the game is sitelocked:
```
            if(urlArg.indexOf("maxgames") != -1) {
                waiting();
            } else if(urlArg.indexOf("file") != -1) {
                waiting();
            } else if(urlArg.indexOf("gywg") != -1) {
                waiting();
            }
```
(from the constructor of `PreloaderLogo`)
The game should only work when running from a local file or on a site with either `maxgames` or `gywg` in its URL.

**Comment by thenickman100** [comment]
@Lord-McSweeney Thanks for your help.  That fixed the issue!  For some reason I didn't think it was a site lock issue.
